### DISCUSSION TITLE
Duplicate Action

### DISCUSSION THREAD

**Discussion by kirkey** [fretje/heroplate#5] [Q&A]
How do the Duplicate Action works?
Thanks in Advance.

**Comment by fretje** [comment]
I suppose you're talking about the `getDuplicateFunc` parameter of the EntityTableContext?

When you specify one of those, there will be an extra "duplicate" button in the actions column, which will open the "create new entity" dialog (AddEditModal), but with all properties of the entity you're duplicating already filled out.
The getDuplicateFunc should do the actual "copying" of all the values you want to duplicate from the "entityToDuplicate" (which is supplied as a parameter) to the new entity. The new entity should be returned from that function.

So for instance the most straightforward way to implement this is:

```c#
        Context = new(
            ...
            getDuplicateFunc: entityToDuplicate =>
                {
                    var newEntity = entity.Adapt<UpdateEntityRequest>();
                    return Task.FromResult(newEntity);
                },
            ...);
```

But you could also individually copy just the fields you need. Or otherwise use `Adapt` and then afterwards empty the fields you don't want to be copied.

**Reply by kirkey** [reply]
thank you for this. I am using FSH API and WASM as my production project base and planning to integrate this function.
